src/openforms/forms/tests/test_translations_converter.py

```diff
@@ -0,0 +1,263 @@
+from django.test import SimpleTestCase
+
+from glom import glom
+from hypothesis import given, settings
+
+from openforms.formio.service import iter_components
+from openforms.formio.tests.search_strategies import (
+    address_nl_component,
+    any_component,
+    bsn_component,
+    checkbox_component,
+    columns_component,
+    content_component,
+    cosign_v1_component,
+    cosign_v2_component,
+    currency_component,
+    date_component,
+    datetime_component,
+    edit_grid_component,
+    email_component,
+    fieldset_component,
+    file_component,
+    iban_component,
+    license_plate_component,
+    map_component,
+    np_family_members_component,
+    number_component,
+    password_component,
+    phone_number_component,
+    postcode_component,
+    radio_component,
+    select_component,
+    selectboxes_component,
+    signature_component,
+    textarea_component,
+    textfield_component,
+    time_component,
+)
+from openforms.formio.typing import (
+    ColumnsComponent,
+    Component,
+    ContentComponent,
+    RadioComponent,
+    SelectBoxesComponent,
+    SelectComponent,
+)
+
+from ..fd_translations_converter import process_component_tree
+
+
+def _get_dummy_translations_store(component):
+    translations_store = {"nl": {}}
+    for comp in iter_components({"components": [component]}, recursive=True):
+        if literal := comp.get("label"):
+            translations_store["nl"][literal] = "Vertaald label"
+    return translations_store
+
+
+def do_processing(component):
+    translations_store = _get_dummy_translations_store(component)
+    process_component_tree([component], translations_store)
+
+
+class ConverterRobustnessTests(SimpleTestCase):
+    """
+    Generate fixtures with hypothesis to check that our converter doesn't unexpectedly
+    crashes on possibly weird but valid data.
+    """
+
+    def assertLabelsTranslated(self, component):
+        for comp in iter_components({"components": [component]}, recursive=True):
+            if not comp.get("label"):
+                continue
+            translation = glom(comp, "openForms.translations.nl.label", default="")
+            self.assertEqual(translation, "Vertaald label")
+
+    @given(any_component())
+    # recursive structure, more expensive to draw examples
+    @settings(max_examples=25, deadline=500)
+    def test_any_component_is_properly_processed(self, component):
+        do_processing(component)
+
+        self.assertLabelsTranslated(component)
+
+    @given(textfield_component())
+    def test_textfield_component(self, component: Component):
+        do_processing(component)
+
+        self.assertLabelsTranslated(component)
```

Comment:
Couldn't this be done in a reusable way, maybe using `subTest`?

Reply:
I wanted:

* a guarantee that every supported component type goes through CI
* possible combinations of optional/required fields for each component type does not cause problems

The second bullet is not exhaustive yet, but using hypothesis we can at least get closer to that goal. I think I tried using hypothesis with subtest and it didn't work, I'm seeing some more info [here](https://github.com/HypothesisWorks/hypothesis/issues/1071#issuecomment-357687573) but it's not conclusive whether subtest is supported or not.

That being said, debugging individual failures with 30 subtests is a pain, while I can now run `src/manage.py test src -k test_textfield_component` to only run that test. I find the DX of this approach much better and easier to follow.